src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/IsMatch.cs

```diff
@@ -41,19 +42,12 @@ public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
             yield return new[] { TexlStrings.IsMatchArg1, TexlStrings.IsMatchArg2, TexlStrings.IsMatchArg3 };
         }
 
-        public override bool CheckTypes(CheckTypesContext context, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        public override void CheckSemantics(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors)
         {
-            bool fValid = base.CheckTypes(context, args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);            
-
-            TexlNode regExNode = args[1];
-            
-            if ((argTypes[1].Kind != DKind.String && argTypes[1].Kind != DKind.OptionSetValue) || !BinderUtils.TryGetConstantValue(context, regExNode, out var nodeValue))                
+            if ((argTypes[1].Kind != DKind.String && argTypes[1].Kind != DKind.OptionSetValue) || !binding.IsConstant(args[1]))
```

Comment:
>argTypes[1] [](http://example.com/codeflow?start=17&length=11)

are we sure that argTypes is not null and length is more than 1? should we check before refer index 1?

Reply:
The function arity (defined in the constructor) defines that it should have at least 2 (and at most 3) arguments, so argTypes.Length is guaranteed to be between 2 and 3.